Classes/NodeCreationHandler/CreationDialogPropertiesCreationHandler.php

```diff
@@ -27,37 +25,22 @@ class CreationDialogPropertiesCreationHandler implements NodeCreationHandlerInte
 {
     /**
      * @Flow\Inject
-     * @var PropertyMapper
+     * @var NodePropertyConversionService
      */
-    protected $propertyMapper;
+    protected $nodePropertyConversionService;
 
-    /**
-     * @param NodeInterface $node The newly created node
-     * @param array $data incoming data from the creationDialog
-     * @return void
-     * @throws PropertyException | SecurityException
-     */
     public function handle(NodeInterface $node, array $data): void
     {
-        $propertyMappingConfiguration = $this->propertyMapper->buildPropertyMappingConfiguration();
-        $propertyMappingConfiguration->forProperty('*')->allowAllProperties();
-        $propertyMappingConfiguration->setTypeConverterOption(PersistentObjectConverter::class, PersistentObjectConverter::CONFIGURATION_OVERRIDE_TARGET_TYPE_ALLOWED, true);
-
-        foreach ($node->getNodeType()->getConfiguration('properties') as $propertyName => $propertyConfiguration) {
+        foreach ($data as $propertyName => $propertyValue) {
```

Comment:
i think this is dangerous to now loop of $data, as this might $somehow contain entries not from the `creationDialog.elements` configuration, like in [Psmb.SplitAdd](https://github.com/psmb/Psmb.SplitAdd/tree/master) were we do some magic like https://github.com/psmb/Psmb.SplitAdd/blob/ceb3d9c3d98755cce1177355e0585e684faa9512/Classes/PropertiesNodeCreationHandler.php#L17

so it should be safer to loop over the configured properties like before and check if its promoted via `showInCreationDialog`

Reply:
This is prevented in 47/37 by continuing if it#s not allowed.

Reply:
ahh i see but actually not really yet. Line 35 expects the property to exist which is unsafe even for legal other creation dialog elements so this access as to be `??` first.

i would argue not to change much of the codeflow in this bugfix and keep looping over `$node->getNodeType()->getConfiguration('properties')` so the actual part of this fix becomes more clear.

For neos 9 i prepared a little refactoring that will cleanup this mess completely by deserialising the elements beforehand

Reply:
> i think this is dangerous to now loop of $data, as this might $somehow contain entries not from the creationDialog.elements configuration

No, now it will *only* loop over what was submitted, because there is no need to loop over all properties instead. Right? Not sure I understand your reference to Psmb.SplitAdd…

But you are right, *if* the configuration is not set, this would break, so `??` makes sense.

Reply:
yeah no fine with me lets loop over $data WITH the null check in place ;)
`Psmb.SplitAdd` is just a random case where we will have 100% a key in $data that is not part of the schema.

Reply:
adjusted ;)